_posts/2020-10-02-nlp-translation.md

```diff
@@ -0,0 +1,205 @@
+  ![test]({{ site.baseurl }}public/images/blog/2020-10-02-nlp-translation/task4b_times.png)
+  *Step times for Transformer*
+</a>
+
+As expected, we can see that compute steps take less time as we increase the number of nodes,
+while communication increasingly takes more and more time, following a logarithmic path. Interestingly, the Transformer model's communication times quickly reach a plateau
+after 4 workers, while GNMT's communication times keeps increasing. This effect is probably due to larger values in the shared tensors. 
+
+Time spent optimizing doesn’t seem to follow the same path, but increases are insignificant (~10 seconds), 
+and are due to additional compute steps (averaging tensors, computations related to Mixed precision) when using distribution.
+
+### Performance comparison
+
+Finally, the following figures show the loss evolution (left), Ratio of communication to total time (center), and a price index (right), 
+ computed as follows $$ index = \frac{price\_increase}{performance\_increase} $$
+
+#### LSTM
+<a href="{{ site.baseurl }}public/images/blog/2020-10-02-nlp-translation/task4a_loss_ratio_prices.png" data-lightbox="task4a_loss_ratio_prices" data-title="Step times for GNMT">
+  ![test]({{ site.baseurl }}public/images/blog/2020-10-02-nlp-translation/task4a_loss_ratio_prices.png)
+  *Step times for GNMT*
+</a>
+
+Communication takes up a huge part of training as we increase distribution:  over 70% of the time is spent sending tensors for 16 workers. 
+This could be made faster by using a more appropriate connectivity between the workers (currently it is at 10GB/s) that can reduce times by a factor of 10 or more.
+An interesting thing to observe is that the curve of cost index first decreases and has a valley before increasing again, which depicts the limits of distribution for this task. 
+The price to performance increase seems to be the best for 4 workers, but all indices are lower than 1, meaning the cost compromise is worth it for this task.
+
+
+#### Transformer
+<a href="{{ site.baseurl }}public/images/blog/2020-10-02-nlp-translation/task4b_loss_ratio_prices.png" data-lightbox="task4b_loss_ratio_prices" data-title="Step times for Transformer">
+  ![test]({{ site.baseurl }}public/images/blog/2020-10-02-nlp-translation/task4b_loss_ratio_prices.png)
+  *Step times for Transformer*
+</a>
+
+Compared to the LSTM model, the communication time ratio is slighly lower, but follows a similar path. For 16 workers, it reaches above 60%. However, the price index has a different shape:
+the observed valley is missing, and the indices are decreasing as we add workers. This suggests a very good performance increase, with a lower price increase. The best configuration 
+according to this index is with 8 workers, but the 16 worker case still has very impressive advantages.
+
+-----
+
+## References
+
+
+{% bibliography --cited %}
```

Comment:
Additionally to the results that are already here, I would find it interesting to compare the computation time to communication time ratio at a fixed 'distribution level' between transformers and LSTMs. I guess that quantity determines how well it scales.